client-react/src/ApiHelpers/ArmHelper.ts

```diff
@@ -211,6 +216,187 @@ const MakeArmCall = async <T, U = T>(requestObject: ArmRequestObject<U>): Promis
   return retObj;
 };
 
+const pollForCompletion = <T, U = T>(response: ArmBatchObject, request: ArmRequestObject<U>) => {
+  const location = getHeader('location', response.headers);
+  const azureAsyncOperation = getHeader('Azure-AsyncOperation', response.headers);
+
+  if (location) {
+    return pollLocationForCompletion(response, location, request);
+  } else if (azureAsyncOperation) {
+    return pollAzureAsyncOperationForCompletion(response, azureAsyncOperation, request);
+  } else if ((<any>response.content)?.properties.provisioningState) {
+  } else {
+    const responseSuccess = response?.httpStatusCode < 300;
+    return Promise.resolve({
+      data: response.content,
+      metadata: {
+        success: responseSuccess,
+        status: response?.httpStatusCode,
+        headers: response?.headers,
+        error: responseSuccess ? null : response.content,
+      },
+    });
```

Comment:
`response` is declared non-nullable in line 219.
```suggestion
    const responseSuccess = response.httpStatusCode < 300;
    return Promise.resolve({
      data: response.content,
      metadata: {
        success: responseSuccess,
        status: response.httpStatusCode,
        headers: response.headers,
        error: responseSuccess ? null : response.content,
      },
    });
```